pyexcelerate/Worksheet.py

```diff
@@ -287,3 +288,9 @@ def get_xml_data(self):
                     style = None
                 row_data.append(self.__get_cell_data(cell, x, y, style))
             yield x, row_data
+
+    def get_auto_filter_xml_string(self):
+        if self.auto_filter:
+            return '<autoFilter ref="{}"/>'.format(
+                Range.Range((1, 1), (self.num_rows, self.num_columns), self)
```

Comment:
Is `<autoFilter>` with a non-encompassing range valid? It seems like it may be better to have the `auto_filter` property take a `Range` or `boolean` type, with the latter setting it as the entire worksheet. My guess is a `@property` and `@auto_filter.setter` will end up being cleaner.

Reply:
To be honest I do not know the precise specification of what is allowable, I simply got this from reversing what Excel did when toggling on the feature. I'll investigate when I get a chance and get back to you.

Reply:
Ok so I took a look at the [open office spec](https://www.ecma-international.org/publications/standards/Ecma-376.htm) regarding `<autoFilter>`, turns out it can take on quite a few values, including have child elements to support more complex filtering.

Your use case is probably the most common though, and I'm assuming you'd like to have it in the lib, so we can merge the PR as-is and we'll take a look at adding support for all the other stuff in a future change. :)

Reply:
Thank you very much @kevmo314  👍